src/utils/polling.js

```diff
@@ -0,0 +1,85 @@
+const { loggers } = require('@asymmetrik/node-fhir-server-core');
+const mkFhir = require('fhir.js');
+const config = require('config');
+const db = require('../storage/DataAccess');
+const topiclist = require('../../public/topiclist.json');
+
+const logger = loggers.get('default');
+const SUBSCRIPTION = 'subscriptions';
+const fhirClientConfig = config.fhirClientConfig;
+
+const TOPIC_URL = 'http://hl7.org/fhir/uv/subscriptions-backport/StructureDefinition/backport-topic-canonical';
+
+/**
+ *
+ * @param {String} topic - Subscription topic from topiclist
+ * @returns {Object} - returns a FHIR search query object
+ */
+function getSearchQuery(topic) {
+  if (topic.includes('encounter')) { return { type: 'Encounter' }; }
+  if (topic.includes('diagnosis')) { return { type: 'Condition' }; }
+  if (topic.includes('medication')) { return { type: 'Medication' }; }
+  if (topic.includes('labresult')) { return { type: 'Observation', query: { category: 'laboratory' } }; }
+  if (topic.includes('order')) { return { type: 'ServiceRequest' }; }
+  if (topic.includes('procedure')) { return { type: 'Procedure' }; }
+  if (topic.includes('immunization')) { return { type: 'Immunization' }; }
+  if (topic === 'demographic-change') { return { type: 'Patient' }; }
+}
+
+function pollSubscriptionTopics() {
+  logger.info('Polling Subscription topics');
+
+  // Get subscriptions with topics
+  const subscriptions = db.select(
+    SUBSCRIPTION,
+    (s) => s.extension && s.extension.some((e) => e.url === TOPIC_URL)
+  );
+
+  // Remove duplicates with Set so we don't poll twice for same topic
+  const topicsToPoll = [
+    ...new Set(
+      subscriptions.map((s) => {
+        const topicExtension = s.extension.find((e) => e.url === TOPIC_URL);
+
+        return topiclist.parameter.find(p => p.valueCanonical === topicExtension.valueUri).name;
+      })
+    ),
+  ];
+
+  if (topicsToPoll.length === 0) {
+    logger.info('No subscription topics to poll.');
+    return;
+  }
+
+  topicsToPoll.forEach((topic) => {
+    logger.info(`Polling EHR for ${topic}.`);
+    const options = {
+      baseUrl: fhirClientConfig.baseUrl,
+      auth: {
+        bearer: 'admin'
```

Comment:
We're going to need more advanced token logic - we won't always have the admin token available. I was going to say let's make that a different task, but I think really it's just going to be copy & pasting the existing client functions from the backend app. https://github.com/mcode/medmorph-backend/blob/main/src/utils/client.js
Thoughts on doing that here vs a separate task?

Reply:
Before we implement any authorization here I think we should at least quickly discuss since we don't _have_ to use SMART backend auth here between the proxy and the EHR

Reply:
We could skip the full SMART backend auth between the backend service client and the proxy (ie, the two pieces we control so we could implement a behind-the-scenes auth method), but not between the proxy and the EHR. It might seem like we could cache a token from the backend app and use that for the EHR, but the proxy is going to poll "forever" so it needs a way to get a token when one expires

Reply:
Added `client` util to get access token and added `clientId` to config.